### PR TITLE
Moved types to types.ts file

### LINKED ISSUE #116: [Feature]: Move types to types.ts
### To which use case does your feature request relate?

_No response_

### Is your feature request related to a problem? Please describe

The types in app.d.ts don't get recognized correctly and no errors are thrown when the structure isn't met. This omits type safety.

### Describe the solution you'd like

Move types to a types.ts, where we can define internal types.

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Slartibartfass2**
Closes #116

## What I have made

- moved types to types.ts to enable type safety and a common location for our types

## Checklist

Either tick or cross out the items that do not apply (using \~\~example text\~\~) and give a reason why the item does not apply.

- ~~[ ] I have updated the documentation accordingly and commented my code~~ (only refactoring)
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~ (only refactoring)
- [x] (for reviewer) I have checked the implementation against the requirements

### CODE REVIEW

File: src/lib/types.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export interface Paper extends PaperSpec {
+    readonly id: number;
+}
+
+export interface PaperSpec {
+    readonly title: string;
+    readonly authors: string[];
+    readonly isBookmarked: boolean;
+}
```
Comment (Merseleo) [inline src/lib/types.ts:9]: Why did you separate paper and paper specification? Is there any use case, where the paper id would have a negative impact or is not existing?

File: src/routes/archivedprojects/+page.svelte [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
 
-    const user = {
-        firstName: "Max",
-        lastName: "Mustermann",
-    };
+    const { data } = $props();
+    const { user } = data;
 </script>
 
 <SimpleNavigationBar {user} title="Archived Projects" backRef="/" tabs={[]} defaultTabValue="" />
```
Comment (Merseleo) [inline src/routes/archivedprojects/+page.svelte:8]: Why do you explicitly create a variable for user here and in the routes/+page.svelte you directly access the user
with user={data.user}?!

Same on following pages ...

**Review by Merseleo — CHANGES_REQUESTED** [review]
Only two questions, but in general a good starting point for the types, thank you!

File: src/lib/types.ts (hunk 1 above)
Comment (Slartibartfass2) [reply src/lib/types.ts:9]: When a new paper is created, there's only the data of the paper but no ID. the paperspec can then be used for the creation of the paper. The same names are used in the backend types, which makes it easier to integrate later.

File: src/routes/archivedprojects/+page.svelte (hunk 2 above)
Comment (Slartibartfass2) [reply src/routes/archivedprojects/+page.svelte:8]: This is just convenience. When the name is the same as the argument, one can only use `{user}`.
We can expect more pattern matching later, i.e. `const { user, paper, ... } = data;`

**Review by Slartibartfass2 — COMMENTED** [review]
Thank you for the review, I answered your comments.

File: src/lib/types.ts (hunk 1 above)
Comment (Merseleo) [reply src/lib/types.ts:9]: Ah ok, makes sense (I forgot the case of creating a paper.

File: src/routes/archivedprojects/+page.svelte (hunk 2 above)
Comment (Merseleo) [reply src/routes/archivedprojects/+page.svelte:8]: Ok, then why did you not follow this pattern in the routes/+page.svelte file?

Comment (Slartibartfass2) [reply src/routes/archivedprojects/+page.svelte:8]: I missed this, good catch, I changed it there.

**Review by Merseleo — APPROVED** [review]
Perfect, thank you for moving the types and adapting everything! Let's merge ;)

**Pull request merged**